cmd/driver/config/config.go

```diff
@@ -169,8 +173,38 @@ func (o *driverConfigOptions) RunDriverConfig(ctx context.Context, cmd *cobra.Co
 	return config.StoreDriver(&driverCfg, o.ConfigFile)
 }
 
+func checkFalcoRunsWithDrivers(engineKind string) error {
+	// Modify the data in the ConfigMap/Falco config file ONLY if engine.kind is set to a known driver type.
+	// This ensures that we modify the config only for Falcos running with drivers, and not plugins/gvisor.
+	// Scenario: user has multiple Falco pods deployed in its cluster, one running with driver,
+	// other running with plugins. We must only touch the one running with driver.
+	if _, err := drivertype.Parse(engineKind); err != nil {
+		return fmt.Errorf("engine.kind is not driver driven: %s", engineKind)
+	}
+	return nil
+}
+
 func replaceDriverTypeInFalcoConfig(hostRoot string, driverType drivertype.DriverType) error {
-	return utils.ReplaceLineInFile(hostRoot+"/etc/falco/falco.yaml", "driver_mode:", "driver_mode: "+driverType.String(), 1)
+	falcoCfgFile := filepath.Join(hostRoot, "etc", "falco", "falco.yaml")
+	type engineCfg struct {
+		Kind string `yaml:"kind"`
+	}
+	type falcoCfg struct {
+		Engine engineCfg `yaml:"engine"`
+	}
+	yamlFile, err := os.ReadFile(filepath.Clean(falcoCfgFile))
+	if err != nil {
+		return err
+	}
+	cfg := falcoCfg{}
+	if err = yaml.Unmarshal(yamlFile, &cfg); err != nil {
+		return err
+	}
+	if err = checkFalcoRunsWithDrivers(cfg.Engine.Kind); err != nil {
+		return err
```

Comment:
I am not sure whether we should error out in this case.
ConfigMap loop does just `continue`.

Reply:
I think we might not want to give an error in this case but just print a warning?
Indeed, we might want to add a `-f,--force` option to force the update.

Reply:
I added the changes in the latest commit.
Example output when running fine and when running with the warning:
```
sudo ./falcoctl driver config --host-root "." --type ebpf
2023-11-24 12:20:02 INFO  Running falcoctl driver config driver host root: . driver type: ebpf

sudo ./falcoctl driver config --host-root "." --type ebpf
2023-11-24 12:20:09 INFO  Running falcoctl driver config driver host root: . driver type: ebpf
2023-11-24 12:20:09 WARN  Avoid updating Falco configuration config: etc/falco/falco.yaml reason: engine.kind is not driver driven: gvisor
```